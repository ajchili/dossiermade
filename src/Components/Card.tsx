import React, { Component, ReactElement } from "react";

export interface Props {
  backgroundColor?: "light" | "dark";
  content: ReactElement;
  title: string;
}

export default class Card extends Component<Props> {
  render() {
    const { backgroundColor = "light", content, title } = this.props;

    return (
      <div
        className={`uk-card uk-card-body ${
          backgroundColor === "light" ? "uk-card-hover" : "uk-card-default"
        }`}
      >
        <p className="uk-h1">{title}</p>
        {content}
      </div>
    );
  }
}
