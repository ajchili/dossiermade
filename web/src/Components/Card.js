import React, { Component } from "react";

class Card extends Component {
  render() {
    const { backgroundColor, content, title } = this.props;

    return (
      <div
        className={`uk-card uk-card-body ${
          backgroundColor === "light" ? "uk-card-hover" : "uk-card-default"
        }`}
      >
        <p className="uk-h1">{title}</p>
        {content}
      </div>
    );
  }
}

Card.defaultProps = {
  backgroundColor: "light",
  title: "Title",
  content: (
    <p>
      Dossier Made a collective media production group based between Washington,
      D.C. and Baltimore, Maryland. We have the resources and availability to
      expand from metropolitan Philidelphia to Southern Virginia. Arrangements
      can also be made to travel further along the east coast and to other North
      American Locations.
    </p>
  )
};

export default Card;
