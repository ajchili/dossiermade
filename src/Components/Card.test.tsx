import * as React from "react";
import Card from "./Card";
import { shallow } from "enzyme";

describe("Card", () => {
  
  it("renders a provided title and content", () => {
    const tree = shallow(<Card title="Title" content={<h3>Content</h3>} />);
    expect(tree.html()).toMatchSnapshot();
  });
  
  it("renders a light card by default", () => {
    const tree = shallow(<Card title="" content={<div/>} />);
    expect(tree.html()).toMatchSnapshot();
  });
  
  it("renders a light card when specified", () => {
    const tree = shallow(<Card title="" content={<div/>} backgroundColor="light" />);
    expect(tree.html()).toMatchSnapshot();
  });
  
  it("renders a dark card when specified", () => {
    const tree = shallow(<Card title="" content={<div/>} backgroundColor="dark" />);
    expect(tree.html()).toMatchSnapshot();
  });
});
