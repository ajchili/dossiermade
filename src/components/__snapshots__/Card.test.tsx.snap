// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders a dark card when specified 1`] = `
<div
  className="uk-card uk-card-body uk-card-default"
>
  <p
    className="uk-h1"
  />
  <div />
</div>
`;

exports[`Card renders a light card by default 1`] = `
<div
  className="uk-card uk-card-body uk-card-hover"
>
  <p
    className="uk-h1"
  />
  <div />
</div>
`;

exports[`Card renders a light card when specified 1`] = `
<div
  className="uk-card uk-card-body uk-card-hover"
>
  <p
    className="uk-h1"
  />
  <div />
</div>
`;

exports[`Card renders a provided title and content 1`] = `
<div
  className="uk-card uk-card-body uk-card-hover"
>
  <p
    className="uk-h1"
  >
    Title
  </p>
  <h3>
    Content
  </h3>
</div>
`;
